### PR TITLE
Actor scheduling without priorities

### PR DESCRIPTION
**Pull request by DougGregor**
The actor runtime has some known issues with deadlock when an actor has
to give up its thread because it's running lower-priority work. To
avoid deadlocks here, disable all of the logic that tries to give up
higher-priority threads when only lower-priority work is available, or
to escalate work, effectively making the actor runtime ignore
priorities internally.

Fixes rdar://79378762.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test macOS

File: stdlib/public/Concurrency/TaskPrivate.h [hunk 1]
```diff
@@ -236,7 +236,7 @@ class alignas(sizeof(void*) * 2) ActiveTaskStatus {
   ActiveTaskStatus withEscalatedPriority(JobPriority priority) const {
     assert(priority > getStoredPriority());
     return ActiveTaskStatus(Record,
-                            (Flags & PriorityMask)
+                            (Flags & ~PriorityMask)
```
Comment (ktoso) [inline stdlib/public/Concurrency/TaskPrivate.h:239]: ❗ nice spotting that

File: stdlib/public/Concurrency/Task.swift [hunk 2]
```diff
@@ -269,7 +269,7 @@ extension Task where Success == Never, Failure == Never {
       }
 
       // Otherwise, query the system.
-      return TaskPriority(rawValue: UInt8(0))
+      return TaskPriority(rawValue: UInt8(_getCurrentThreadPriority()))
```
Comment (ktoso) [inline stdlib/public/Concurrency/Task.swift:272]: 👍

**Review by kavon — APPROVED** [review]

**Review by rjmccall — APPROVED** [review]
LGTM

**Pull request merged**